code.go

```diff
@@ -0,0 +1,260 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+	"unsafe"
+)
+
+// 全局变量命名不规范
+var global_var int
+var GlobalVar2 string
+
+// 常量命名错误
+const max_SIZE = 100
+const minsize = 10
+
+// 1. 内存泄漏 - goroutine泄漏
+func memoryLeak() {
+	ch := make(chan int)
+	go func() {
+		for {
+			select {
+			case <-ch:
+				return
+			default:
+				// 这个goroutine永远不会退出
+				time.Sleep(time.Second)
+			}
+		}
+	}()
+	// 忘记关闭channel
+}
+
+// 2. 竞态条件
+var counter int
+
+func raceCondition() {
+	for i := 0; i < 1000; i++ {
+		go func() {
+			counter++ // 没有锁保护
+		}()
+	}
+}
+
+// 3. 错误的错误处理
+func badErrorHandling() {
+	file, err := os.Open("nonexistent.txt")
+	if err != nil {
+		// 错误：忽略错误
+		return
+	}
+	defer file.Close()
+
+	data := make([]byte, 100)
+	n, err := file.Read(data)
+	// 错误：没有检查错误
+	fmt.Println("Read", n, "bytes")
+}
+
+// 4. 资源泄漏
+func resourceLeak() {
+	file, _ := os.Open("test.txt") // 忽略错误
+	// 错误：没有defer close
+	data := make([]byte, 100)
+	file.Read(data)
+	// file永远不会被关闭
+}
+
+// 5. 切片越界和nil指针
+func sliceAndNilErrors() {
+	var slice []int
+	fmt.Println(slice[0]) // panic: 空切片访问
+
+	var ptr *int
+	fmt.Println(*ptr) // panic: nil指针解引用
+
+	arr := []int{1, 2, 3}
+	fmt.Println(arr[10]) // panic: 索引越界
+}
+
+// 6. map并发读写
+var sharedMap = make(map[string]int)
+
+func mapConcurrencyIssue() {
+	go func() {
+		for {
+			sharedMap["key"] = 1 // 并发写
+		}
+	}()
+
+	go func() {
+		for {
+			_ = sharedMap["key"] // 并发读
+		}
+	}()
+}
+
+// 7. 无限循环
+func infiniteLoop() {
+	for {
+		// 没有退出条件
+		fmt.Println("Running forever...")
+	}
+}
+
+// 8. 不当的类型转换
+func badTypeConversion() {
+	var i int64 = 1000000000000
+	var j int32 = int32(i) // 数据截断
+	fmt.Println(j)
+
+	str := "not a number"
+	num, _ := strconv.Atoi(str) // 忽略转换错误
+	fmt.Println(num)
+}
+
+// 9. 字符串操作性能问题
+func inefficientStringOps() {
+	var result string
+	for i := 0; i < 10000; i++ {
+		result += "a" // 每次都创建新字符串
+	}
+	return
```

Comment:
Use strings.Builder for efficient string concatenation. String concatenation in loop creates O(n²) complexity and excessive allocations.

```suggestion
	var builder strings.Builder
	for i := 0; i < 10000; i++ {
		builder.WriteString("a")
	}
	result := builder.String()
```